plugins/outputs/nats/nats.go

```diff
@@ -85,8 +132,51 @@ func (n *NATS) Connect() error {
 
 	// try and connect
 	n.conn, err = nats.Connect(strings.Join(n.Servers, ","), opts...)
+	if err != nil {
+		return err
+	}
+
+	if n.Jetstream != nil {
+		n.jetstreamClient, err = jetstream.New(n.conn)
+		if err != nil {
+			return fmt.Errorf("failed to connect to jetstream: %w", err)
+		}
+
+		if len(n.Jetstream.Subjects) == 0 {
+			n.Jetstream.Subjects = []string{n.Subject}
+		}
+		if !choice.Contains(n.Subject, n.Jetstream.Subjects) {
+			n.Jetstream.Subjects = append(n.Jetstream.Subjects, n.Subject)
+		}
```

Comment:
Hmmm shouldn't the check be more strict and check the last element only? I mean what happens with
```
subject = "foo"
jetstream.subjects = ["some", "foo", "topic"]
```
which will render your condition false but this certainly not what you intend here, is it?

Reply:
the idea is `jetstream.subjects` should contain the subject present in the outer configuration. 
so if its not present we append it, since ultimately the subject is used to publish the message.

so in your example, as you pointed out the condition will be false and subject wont be appended.

Reply:
that is correct. we just need the subject present, it can be any where in the array. I have added the comment as suggested.